ext/fg/js/frontend.js

```diff
@@ -262,7 +371,97 @@ class Frontend {
 
         this.textSourceLast = null;
     }
+
+    getPrimaryTouch(touchList) {
+        return touchList.length > 0 ? touchList[0] : null;
+    }
+
+    getIndexOfTouch(touchList, identifier) {
+        for (let i = 0, ii = touchList.length; i < ii; ++i) {
+            let t = touchList[i];
+            if (t.identifier === identifier) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    excludeTouches(touchList, excludeTouchList) {
+        const result = [];
+        for (let i = 0, ii = touchList.length; i < ii; ++i) {
+            let r = result[i];
```

Comment:
Is this supposed to be `let r = touchList[i]`? You'd be getting `undefined` here. If that's the case, you should also use the `for` `of` loop instead of manually doing indexing.

Reply:
You are correct; updated.